### ISSUE TITLE
Trying to replace a parameter and replacing the wrong "?"

### ISSUE THREAD

**Issue by romulets** [metabase/metabase#7325]
Environment:
- Browser: Chrome Version 65.0.3325.162 (Official Build) (64-bit)
- OS : Ubuntu 16.04 LTS
- DB: Google BigQuery
- Metabase Version: v0.28.3

I'm performing a query which uses regular expressions to extract texts and uses metabase parameters to filter my data. Inside my regular expression I use the operator "?" to retrieve specific information, so I have something similar to this:

```
create temporary function extract(text)
  returns string 
  as ((  select regexp_extract(text, '(?: prefix )(.*?)(?: suffix)') ));

select 
  extract(subject) 
from ds.table 
[[ where date >= {{ date_filter }} ]]; 
```

After I run my query (filling the parameter), the query which metabase ran is:

```
create temporary function extract(text)
  returns string 
  as ((  select regexp_extract(text, '((timestamp('2018-04-03T00:00:00.000Z'): prefix )(.*?)(?: suffix)') ));

select 
  extract(subject) 
from ds.table 
[[ where date >= ? ]]; 
```

As we can see, it's replacing the wrong character "?". 

Is it a bug? Does it have a workaround?

**Comment by camsaul** [comment]
`?` is the normal placeholder for parameterized SQL queries in JDBC. Our code that generates SQL queries generates parameterized queries, but since BigQuery doesn't support them, we have extra logic swap out `?` placeholders with appropriate literals. So putting extra `?`s in your query is liable to break things with the way our code currently works.

It might be a while before we're able to fix the issue so in the meantime I would suggest using equivalent functions if they're available

**Comment by flamber** [comment]
This looks like a duplicate of #5091

**Comment by flamber** [comment]
Fixed in 0.35.4
